docs/support/tutorials/ml-llm.md

```diff
@@ -180,33 +180,101 @@ Face](https://huggingface.co/blog/mlabonne/sft-llama3), which also
 covers the [Unsloth library](https://github.com/unslothai/unsloth).
 
 
-## Using quantization
+## Quantization
 
-Using the `bitsandbytes` library, you can also use 4-bit
-quantization. [Quantization has been integrated into Hugging Face
-Transformers as
-well](https://huggingface.co/blog/4bit-transformers-bitsandbytes).
+Quantization is a process that converts the weights and activations within an LLM from high-precision values, such as 32-bit floating-point, to lower-precision ones, such as an 8-bit integer. This leads to a significant decrease in overall model size, leading to smaller memory needs with a slight drop in accuracy. You can learn more about quantization for example in [this tutorial](https://www.datacamp.com/tutorial/quantization-for-large-language-models).
 
+Quantization can be done during inference or training phase. [Post-Training Quantization (PTQ)](https://docs.pytorch.org/TensorRT/ts/ptq.html) involves quantizing a pre-trained model during the inference phase. [Quantization-Aware Training (QAT)](https://www.ibm.com/think/topics/quantization-aware-training) is applied during training to simulate the effects of quantization, resulting in a model more robust to quantization noise. This guide focuses on Post-Training Quantization.
+
+It can take hours to quantize very large models, but luckily many models already have a quantized version available, for example in Hugging Face. You can look for quantized models by a suffix in the model name indicating a quantization method, such as AWQ, GPTQ, or GGUF, or alternatively, model precision, such as 8bit or 4bit. 
+
+### Using BitsandBytes Quantization
+
+The [BitsAndBytes library from Hugging Face Transformers](https://huggingface.co/docs/transformers/en/quantization/bitsandbytes) offers runtime quantization, which compresses weights to 8-bit or 4-bit on-the-fly during inference to save memory.
 
 ```python
-from transformers import BitsAndBytesConfig
+from transformers import BitsAndBytesConfig, AutoModelForCausalLM
+
 bnb_config = BitsAndBytesConfig(
-    load_in_4bit=True,
-    bnb_4bit_quant_type="nf4",
-    bnb_4bit_compute_dtype=torch.bfloat16,
-    bnb_4bit_use_double_quant=True,
-    bnb_4bit_quant_storage=torch.bfloat16,
+   load_in_4bit=True,
+   bnb_4bit_quant_type="nf4",
+   bnb_4bit_compute_dtype=torch.bfloat16,
+   bnb_4bit_use_double_quant=True,
+   bnb_4bit_quant_storage=torch.bfloat16,
+)
+
+model = AutoModelForCausalLM.from_pretrained(
+   args.model,
+   quantization_config=bnb_config,
+   ...
+)
+```
+You can use it in our fine-tuning LLMs example (see section above) with the --4bit argument. Alternatively, see our [Github repository](https://github.com/CSCfi/llm-quantization-scripts/tree/main/BitsAndBytes) for a full example to quantize a model using bitsandbytes on Puhti, Mahti or LUMI.
```

Comment:
maybe a command line flags looks nicer with the "code" font, like this: `--4bit`